include/device.h

```diff
@@ -208,6 +216,34 @@ extern "C" {
 
 struct device;
 
+/**
+ * @brief Device PM info
+ *
+ * @param dev pointer to device structure
+ * @param lock lock to synchronize the get/put operations
+ * @param pm_lock lock to synchronize the PM state transitions
+ * @param enable device pm enable flag
+ * @param usage device usage count
+ * @param async_request device pm request
+ * @param async_req_status device pm request status
+ * @param pm_status device idle pm status
+ * @param dwork device pm work for asyncroneous request
+ * @param timeout delayed work timeout value in msec
+ * @param signal signal to notify the Async API callers
+ */
+struct device_pm {
+	struct device *dev;
+	struct k_sem lock;
+	struct k_sem pm_lock;
+	bool enable;
+	s32_t usage;
+	u32_t async_request;
+	u32_t async_req_status;
+	u32_t pm_status;
+	struct k_delayed_work dwork;
+	u32_t timeout;
+	struct k_poll_signal *signal;
```

Comment:
Could we embed signal here? If we do so, driver will not need to implements its own.